src/Build/Construction/ProjectRootElement.cs

```diff
@@ -58,10 +58,18 @@ public class ProjectRootElement : ProjectElementContainer
 
         private static readonly ProjectRootElementCacheBase.OpenProjectRootElement s_openLoaderPreserveFormattingDelegate = OpenLoaderPreserveFormatting;
 
+        private const string XmlDeclarationPattern = @"\A\s*\<\?\s*xml.*\?\>\s*\Z";
+
         /// <summary>
         /// Used to determine if a file is an empty XML file if it ONLY contains an XML declaration like &lt;?xml version="1.0" encoding="utf-8"?&gt;.
         /// </summary>
-        private static readonly Lazy<Regex> XmlDeclarationRegEx = new Lazy<Regex>(() => new Regex(@"\A\s*\<\?\s*xml.*\?\>\s*\Z"), isThreadSafe: true);
+#if NET
+        [GeneratedRegex(XmlDeclarationPattern)]
+        private static partial Regex XmlDeclarationRegex { get; }
+#else
+        private static Regex XmlDeclarationRegex => s_xmlDeclarationRegex ??= new Regex(XmlDeclarationPattern);
```

Comment:
Sorry @simatian https://github.com/dotnet/msbuild/pull/11228#discussion_r1906144062

Reply:
While I was responsible for introducing Lazy, it's often not the right answer, in particular because you can often get the laziness you need without needing to create the Lazy itself.

In this case, we don't rely on the object identity of the Regex, so we don't need to worry about race conditions to publish an instance; worse case is two concurrent uses will end up using different instances.

My assumption is it'll be relatively rare for there to be such concurrent use on initialization (e.g. two threads in the same process trying to use this property for the first time within the same, say, millisecond), in which case we can avoid paying for the Lazy itself.